flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RelDataTypeJsonDeserializer.java

```diff
@@ -176,14 +173,32 @@ private RelDataType deserialize(JsonNode jsonNode, FlinkDeserializationContext c
                                 : null;
                 final RelDataType type;
                 if (sqlTypeName == SqlTypeName.ARRAY) {
-                    RelDataType elementType = deserialize(objectNode.get(FIELD_NAME_ELEMENT), ctx);
+                    RelDataType elementType =
+                            deserialize(
```

Comment:
This makes the code quite verbose. Now `.traverse(jsonParser.getCodec())` is everywhere in the code. Isn't it possible to restore the original code at this location?